theta/src/Theta/Target/Avro/Types.hs

```diff
@@ -294,9 +294,12 @@ record recordName avroVersion doc Fields { fields } = do
         , Avro.fldDoc     = getText <$> fieldDoc
         , Avro.fldOrder   = Nothing
         , Avro.fldType    = type_
-        , Avro.fldDefault = Nothing
+        , Avro.fldDefault = defaultValue fieldType type_
         }
 
+    defaultValue Type{baseType = Optional' _} Avro.Union{options} = Just (Avro.DUnion options Avro.Null Avro.DNull)
```

Comment:
This alignment relies on the function below always returning a `Union` schema for a base type of `Optional'`. That seems to always be the case, however, so I'm kind of relying on that.